src/components/RegisterForm.vue

```diff
@@ -0,0 +1,190 @@
+
+    <button @click.prevent="submit"> Submit</button> <br><br> 
+    <hr size="3" noshade>
+  
+  </div>
+</template>
+
+<script>
+export default {
+  data: function() {
+    return {
+      fname: '',
+      mail: '',
+      phonenum: '',
+      gender: '',
+      image: '',
+      msg: 'Register Here',
+      imgs:[
+        {src: require("../assets/icons/dog.png") , value:"dog"},
+        {src: require("../assets/icons/fish.png"), value:"fish"},
+        {src: require("../assets/icons/owl.png"), value:"owl"}
+      ]
+    };
+  },
+  methods: {
+    submit: function(){
+      if(this.emailValidation && this.phonenumValidation&&
+         this.fnameValidation && this.genderValidation&&
+         !this.imgNoValidation)
+      {
+        alert('submitted');
+        console.log(this.jsonOutput);
+        this.cleanForm()
+      }
+      else{
+        alert('oops');
+      }
+    },
+    cleanForm: function(){
+      this.fname = '';
+      this.mail = '';
+      this.phonenum = '';
+      this.gender = '';
+      this.image = '';
+    }
+  },
+  computed: {
+    // fnameCaps: function() {
+    //   return this.fname.toUpperCase();
+    fnameValidation: function(){
+      if(this.fname.match(/^([a-zA-Z]{1,} )([a-zA-Z]{1,})$/))
+        return true;
+      else
+        return false;
+    },
+    emailValidation: function () {
+      if (this.mail.match(/^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/)) {
+        return true;
+      }
+      else{
+       return false;
+      }
+    },
+    phonenumValidation: function(){
+      if (this.phonenum.match(/^\d{10}$/)) {
+        return true;
+      }
+      return false;
+    },
+    genderValidation: function(){
+      if(this.gender != '')
+        return true;
+      else
+        return false;
+    },
+    imgNoValidation: function(){
+      if(this.image == '')
+        return true;
+      else
+        return false;
+    },
+    jsonOutput: function(){
+      return [
+        ['fname', this.fname],
+        ['mail', this.mail],
+        ['phonenum', this.phonenum],
+        ['gender', this.gender],
+        ['image', this.image]
+      ];
+    },
+    emptyFname: function(){
+      return this.fname == '';
+    },
+    emptyMail: function(){
+      return this.mail == '';
+    },
+    emptyPhoneNumber: function(){
+      return this.phonenum == '';
+    }
+  }
+}
+</script>
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<style scoped>
+h2 {
+  font-style: italic;
+}
+
+h1 {
+  font-style: italic;
+  text-decoration-line:underline;
+  text-decoration-style:dotted;
+}
+
+.Valid {
+  border-color:rgb(3, 245, 3);
+}
+
+.NoValid {
+  border-color:red;
+}
+p{
+  color: red;
+}
+
+[type=radio] {
+  display: none;
+}
+
+[type=radio]+img {
+  cursor: pointer;
+}
+
+[type=radio]:checked+img {
+  outline: 2px solid rgb(8, 192, 24);
+}
+
+button {
+  border:double;
+  font-size:larger;
+}
+
+/* additional for the v-cloak at the beginning of the form */
```

Comment:
no need for that comment - again, we only need comments when expecting another programmer to understand this code just by reading it is unreasonable, in any other case were the code is fairly simple there is no need for such comments